client/matching/peerResolver.go

```diff
@@ -29,22 +29,25 @@ import (
 // Those are deployed instances of Cadence matching services that participate in the cluster ring.
 // The resulting peer is simply an address of form ip:port where RPC calls can be routed to.
 type PeerResolver struct {
-	resolver      membership.Resolver
-	addressMapper AddressMapperFn
+	resolver  membership.Resolver
+	service   string
+	namedPort string // grpc or tchannel, depends on yarpc configuration
 }
 
-type AddressMapperFn func(string) (string, error)
-
 // NewPeerResolver creates a new matching peer resolver.
-func NewPeerResolver(membership membership.Resolver, addressMapper AddressMapperFn) PeerResolver {
-	return PeerResolver{membership, addressMapper}
+func NewPeerResolver(membership membership.Resolver, namedPort string) PeerResolver {
+	return PeerResolver{
+		resolver:  membership,
+		namedPort: namedPort,
+		service:   service.Matching,
+	}
 }
 
 // FromTaskList resolves the matching peer responsible for the given task list name.
 // It uses our membership provider to lookup which instance currently owns the given task list.
 // FromHostAddress is used for further resolving.
 func (pr PeerResolver) FromTaskList(taskListName string) (string, error) {
-	host, err := pr.resolver.Lookup(service.Matching, taskListName)
```

Comment:
Any reason why this is a struct member now, instead of using `service.Matching` like before? It is still initialized to the same const in the constructor.